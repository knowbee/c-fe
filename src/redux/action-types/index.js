export * from "./registerUserTypes";
export * from "./loginTypes";
export * from "./currentUserTypes";
export * from "./usersTypes";
