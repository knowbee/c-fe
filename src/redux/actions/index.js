export * from "./registerActions";
export * from "./loginActions";
export * from "./currentUserActions";
export * from "./usersActions";
export * from "./chatActions";
