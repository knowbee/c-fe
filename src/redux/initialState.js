module.exports = {
  currentUser: {
    message: "",
    isLoggedIn: localStorage.getItem("token") !== null,
    chats: [],
    messages: [],
    profile: {},
  },
  register: {
    submitting: false,
    message: "",
    email: "",
    name: "",
    password: "",
    errors: {},
  },
  users: {
    fetching: false,
    chats: [],
    errors: {},
  },
  login: {
    submitting: false,
    credentials: {
      email: "",
      password: "",
    },
  },
};
