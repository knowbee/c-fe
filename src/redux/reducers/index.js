import { combineReducers } from "redux";
import register from "./registerReducer";
import login from "./loginReducer";
import currentUser from "./currentUserReducer";
import users from "./usersReducer";
import chat from "./chatReducer";

const reducer = combineReducers({
  register,
  login,
  currentUser,
  users,
  chat,
});
export default reducer;
