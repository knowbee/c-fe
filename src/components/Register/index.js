import React, { useState } from "react";
import { connect } from "react-redux";
import { submitRegister } from "../../redux/actions";
import "./register.scss";
import Layout from "../Layout";
import { useHistory } from "react-router-dom";
const Register = ({ onSubmit }) => {
  const history = useHistory();

  const goToLogin = () => {
    history.push("/login");
  };

  const [name, setName] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const onSubmitButton = (e) => {
    e.preventDefault();
    onSubmit({ name, email, password }).then((res) => {
      if (res.status === 201) {
        history.push("/login");
      }
    });
  };
  return (
    <Layout>
      <div className="registration-form">
        <div className="registration-form-header">
          <img src="https://iili.io/fi0ROQ.png" alt="logo-2" />
          <ul>
            <li>Registration</li>
          </ul>
        </div>
        <form>
          <input
            value={name}
            onChange={(e) => setName(e.target.value)}
            type="text"
            placeholder="Your name"
          ></input>
          <input
            value={email}
            onChange={(e) => setEmail(e.target.value)}
            type="text"
            placeholder="E-mail"
          ></input>
          <input
            value={password}
            onChange={(e) => setPassword(e.target.value)}
            type="password"
            placeholder="Password"
          ></input>
          <button onClick={onSubmitButton} className="registration-button">
            Registration
          </button>
          <button onClick={goToLogin} className="signin-button" type="submit">
            Sign in
          </button>
        </form>
      </div>
    </Layout>
  );
};

export const mapStateToProps = (state) => ({
  name: state.register.name,
  email: state.register.email,
  password: state.register.password,
  errors: state.register.errors,
  submitting: state.register.submitting,
});

export const mapDispatchToProps = (dispatch, props) => ({
  onSubmit: ({ name, email, password }) =>
    dispatch(
      submitRegister({
        name,
        email,
        password,
        props,
      })
    ),
});

export default connect(mapStateToProps, mapDispatchToProps)(Register);
