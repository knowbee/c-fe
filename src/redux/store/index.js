import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension/developmentOnly";
import initialState from "../initialState";
import reducers from "../reducers/index";

const store = createStore(
  reducers,
  initialState,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
