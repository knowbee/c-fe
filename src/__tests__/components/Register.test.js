import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { createStore } from "redux";

import "./index.css";
import rootReducer from "../../redux/reducers";
import Register from "../../components/Register";

const store = createStore(rootReducer);

test("Register page loads with default state", () => {
  const component = renderer.create(
    <Provider store={store}>
      <Register />
    </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
