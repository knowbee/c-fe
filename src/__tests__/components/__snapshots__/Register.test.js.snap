// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register page loads with default state 1`] = `
<div
  className="layout"
>
  <div
    className="left-side"
  >
    <div
      className="middle-container"
    >
      <img
        alt="logo"
        src="https://iili.io/fi0IUb.png"
      />
      <p>
        stay in touch with your friends anytime and anywhere
      </p>
    </div>
  </div>
  <div
    className="right-side"
  >
    <div
      className="registration-form"
    >
      <div
        className="registration-form-header"
      >
        <img
          alt="logo-2"
          src="https://iili.io/fi0ROQ.png"
        />
        <ul>
          <li>
            Registration
          </li>
        </ul>
      </div>
      <form>
        <input
          onChange={[Function]}
          placeholder="Your name"
          type="text"
          value=""
        />
        <input
          onChange={[Function]}
          placeholder="E-mail"
          type="text"
          value=""
        />
        <input
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
        <button
          className="registration-button"
          onClick={[Function]}
        >
          Registration
        </button>
        <button
          className="signin-button"
          onClick={[Function]}
          type="submit"
        >
          Sign in
        </button>
      </form>
    </div>
  </div>
</div>
`;
